dashboard/app/reporting_email.go

```diff
@@ -694,59 +695,120 @@ func handleTestCommand(c context.Context, info *bugInfoResult,
+
+The list of subsystems can be found at %s
+`
 	cmdNoSubsystems = `Subsystems assignment is not yet supported for this kernel.
 Please contact the bot's maintainers.`
 	cmdUnknownSubsystemFmt = `Please use subsystem names from the list of supported subsystems:
 %s
 
 If you believe that the subsystem list should be changed, please contact the bot's maintainers.`
-	cmdNoSubsystemsSetFmt = `You must specify at least one subsystem name.
-The list of available subsystems can be found at %s`
+	cmdInternalErrorReply = `The command was not executed due to an internal error.
+Please contact the bot's maintainers.`
 )
 
-func handleSetCommand(c context.Context, info *bugInfoResult,
-	msg *email.Email, command *email.SingleCommand) string {
-	subsystemsURL := fmt.Sprintf("%v/%v/subsystems?all=true", appURL(c), info.bug.Namespace)
+func handleSetCommand(c context.Context, bug *Bug, msg *email.Email,
+	command *email.SingleCommand) string {
+	subsystemsURL := fmt.Sprintf("%v/%v/subsystems?all=true", appURL(c), bug.Namespace)
 	match := setCmdRe.FindStringSubmatch(command.Args)
-	if len(match) != 3 {
-		return fmt.Sprintf(cmdParseFailureFmt, subsystemsURL)
-	}
-	cmd, args := match[1], match[2]
-	// Now we only support setting bug's subsystems.
-	// Also let's tolerate both subsystem spellings.
-	if cmd != "subsystem" && cmd != "subsystems" {
-		return fmt.Sprintf(cmdParseFailureFmt, subsystemsURL)
-	}
-	service := getSubsystemService(c, info.bug.Namespace)
-	if service == nil {
-		return cmdNoSubsystems
-	}
-	if args == "" {
-		return fmt.Sprintf(cmdNoSubsystemsSetFmt, subsystemsURL)
+	if match == nil {
+		return fmt.Sprintf(setBugCmdFormat, subsystemsURL)
+	}
+	group, labels := match[1], match[2]
+	tags := []BugLabel{}
+	log.Infof(c, "setting %#v tags for %q (group %q)", bug.displayTitle(), labels, group)
+	for _, name := range setCmdArgSplitRe.Split(labels, -1) {
+		tags = append(tags, BugLabel{
+			Name:  name,
+			SetBy: msg.Author,
+			Link:  msg.Link,
+		})
 	}
-	subsystems := []*subsystem.Subsystem{}
-	for _, name := range setCmdArgSplitRe.Split(args, -1) {
-		item := service.ByName(name)
-		if item == nil {
-			return fmt.Sprintf(cmdUnknownSubsystemFmt, subsystemsURL)
+	var err error
+	if group == "subsystem" || group == "subsystems" {
+		// Let's tolerate both subsystem spellings.
+		service := getSubsystemService(c, bug.Namespace)
+		if service == nil {
+			return cmdNoSubsystems
+		}
+		for _, tag := range tags {
+			if service.ByName(tag.Name) == nil {
+				return fmt.Sprintf(cmdUnknownSubsystemFmt, subsystemsURL)
+			}
+		}
+		err = updateSingleBug(c, bug.key(c), func(bug *Bug) error {
+			bug.Tags.Subsystems = tags
+			return nil
+		})
+	} else if group == "" {
+		if tags[0].Name == "subsystem" {
+			// Most likely user has made a mistake.
+			// Display the instructions once more.
+			return fmt.Sprintf(setBugCmdFormat, subsystemsURL)
 		}
-		subsystems = append(subsystems, item)
+		err = updateSingleBug(c, bug.key(c), func(bug *Bug) error {
```

Comment:
> fancy labels people decide to use

Was it not actually the purpose? :)

I'm not sure it'll severely restrict automation. So far it's only `no-reminders`, later there'll also appear `missing backport`. I don't think it's super critical if someone misspells them.

Tree origin labels will be in a separate group. What else could there be that we could later want to automatically process?

Reply:
People will misspell labels for which we already have meaning.
People will want to mark "important" bugs, some will attach "p0", some "high_prio", some "high-prio", "security", "critical", etc. And then it will be impossible to select "important" bugs because all of them are spelled differently (+a bunch of typos) and people will start re-marking bugs for the spelling they use. And then we will end up with huge ugly lists of tags for each bug. And it won't be possible to, say, number of important unfixed bugs.
Same will go for assigned/work-in-progress/wip and just everything else.
We may use some of these in some conditions in the code in future, but the problem is equal for just selecting bugs based on tag manually. If tags are spelled differently, it's impossible to select and they become useless.
Also consider: all global and visible to everyone, they are not private. And thus there should be some common ground and one global spelling for one meaning.